File: src/app/item.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Item } from './item.model';

@Injectable({
  providedIn: 'root'
})
export class ItemService {

  private apiUrl = 'http://localhost:8080/itens';

  constructor(private http: HttpClient) {}

  listar(): Observable<Item[]> {
    return this.http.get<Item[]>(`${this.apiUrl}/listartodos`);
  }

  criar(item: Omit<Item, 'id'>): Observable<Item> {
    return this.http.post<Item>(`${this.apiUrl}/salvar`, item);
  }

  buscarPorId(id: string): Observable<Item> {
    return this.http.get<Item>(`${this.apiUrl}/${id}`);
  }

  atualizar(id: string, item: Item): Observable<Item> {
    return this.http.put<Item>(`${this.apiUrl}/atualizar/${id}`, item);
  }

  excluir(id: string): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

File: GerenciadorItensSustentaveis/src/main/java/com/GerenciadorItensSustentaveis/Service/ItemService.java
package com.GerenciadorItensSustentaveis.Service;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

import com.GerenciadorItensSustentaveis.Model.Item;

public interface ItemService  {
	
	Item salvar(Item item);
	
	Optional <Item> buscarPorId (UUID id);
	
	List<Item> listarTodos();
	
	Item atualizar(UUID id, Item item);
	
	void deletar(UUID id);
	

}
